py/sentry_streams/example_config.py

```diff
@@ -1,16 +1,16 @@
-from sentry_streams.sinks import Pipeline, RawKafkaSink, RawKafkaSource
+from sentry_streams.pipeline import Pipeline, RawKafkaSink, RawKafkaSource
 
 # pipeline: special name
 pipeline = Pipeline()
 
-source = RawKafkaSource(name="myinput", ctx=pipeline, logical_topic="logical-events")
+source = RawKafkaSource(
+    name="myinput", ctx=pipeline, logical_topic="logical-events", step_type="source"
```

Comment:
The idea behind adding `step_type` was to give us a way to map our primitives API (e.g. `RawKafkaSource`) to a canonical primitive name (e.g.`Source`). We'd do this for other primitives too if we like this way of mapping them. 

Open to other ways of mapping as well. I could also use the dataclass names to figure out which runtime primitives they correspond to.

Reply:
Isn't a `RawKafkaSource` always a source as well as a `RawKafkaSink` is always a sink? 
If this is correct and we need to classify whether a step is a sourceor a sink, that should be a structural part of the class rather than an argument to the constructor. 

A few ways to do that are:
- provide a static property for each of the class that returns the type, or, much better, return what is needed by the client to chaneg its behavior depending whether this is a source or a sink/
- Extend a class that contains everything that makes a class a source/sink